File: ServerApp/Entities/Periods/SemesterDepartment.cs
using System.Collections.Generic;
using Everest.AspNetStartup.Binding;
using Everest.AspNetStartup.Models;
using Microsoft.AspNetCore.Mvc;
using Newtonsoft.Json;

namespace Exam.Entities.Periods
{
    
    [ModelBinder(BinderType = typeof(ItemValueModelBinder))]
    public class SemesterDepartment:Entity<long>
    {
        public bool Opened { get; set; }
        public bool Closed { get; set; }
        public bool Pending { get; set; } = true;
        
        [JsonIgnore]
        public virtual YearDepartment YearDepartment { get; set; }
        public long? YearDepartmentId { get; set; }

        [JsonIgnore]
        public virtual Semester Semester { get; set; }
        public long? SemesterId { get; set; }

        [JsonIgnore]
        public virtual List<SemesterLevel> SemesterLevels { get; set; }

        [JsonIgnore]
        public virtual List<SemesterSpeciality> SemesterSpecialities { get; set; }
    }
}

File: webapp/apps/regolia/src/models/entities/semester/semester-department.ts
import {Semester} from "./semester";
import {Department} from "../organisation";
import {Entity} from "../entity";
import {SemesterLevel} from "./semester-level";
import {SemesterSpeciality} from "./semester-speciality";
import {YearDepartment} from "../year";

export class SemesterDepartment extends Entity<number> {

  constructor(value: any = {}) {
    super();

    this.id = value.id;
    this.registrationDate = new Date(value.registrationDate);

    this.semesterId = value.semesterId;
    this.yearDepartmentId = value.yearDepartmentId;
  }

  yearDepartmentId: number;
  yearDepartment: YearDepartment;

  semester: Semester;
  semesterId: number;

  semesterLevels: SemesterLevel[];
  semesterSpecialities: SemesterSpeciality[];

  url(path?: string): string {
    const url = `${this.semester.url()}/departments/${this.id}`;
    if(path) {
      return `${url}/path`;
    }
    return url;
  }
}
